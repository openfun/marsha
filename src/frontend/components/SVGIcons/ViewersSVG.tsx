import React from 'react';

import SVGIcon, { SvgProps } from '.';

export const ViewersSVG = (svgProps: SvgProps) => {
  return (
    <SVGIcon viewBox={'0 0 54 54'} {...svgProps}>
      <g fillRule="evenodd" stroke="none" strokeWidth="1">
        <g transform="translate(2.5 2.5)">
          <path
            fillRule="nonzero"
            d="M25 25c3.396 0 6.396.929 8.833 2.143 2.25 1.143 3.667 3.714 3.667 6.5v1.476c0 1.31-.938 2.381-2.083 2.381H14.583c-1.146 0-2.083-1.071-2.083-2.381v-1.452c0-2.81 1.417-5.381 3.667-6.5C18.604 25.929 21.604 25 25 25zm-14.583 2.083a4.179 4.179 0 004.166-4.166 4.179 4.179 0 00-4.166-4.167 4.179 4.179 0 00-4.167 4.167 4.179 4.179 0 004.167 4.166zm2.354 2.292a14.547 14.547 0 00-2.354-.208c-2.063 0-4.021.437-5.792 1.208a4.19 4.19 0 00-2.542 3.854v1.188A2.09 2.09 0 004.167 37.5h7.291v-3.354c0-1.73.48-3.354 1.313-4.771zm26.812-2.292a4.179 4.179 0 004.167-4.166 4.179 4.179 0 00-4.167-4.167 4.179 4.179 0 00-4.166 4.167 4.179 4.179 0 004.166 4.166zm8.334 7.146c0-1.687-1-3.187-2.542-3.854a14.48 14.48 0 00-5.792-1.208c-.812 0-1.583.083-2.354.208a9.372 9.372 0 011.313 4.77V37.5h7.291a2.09 2.09 0 002.084-2.083v-1.188zM25 10.417a6.242 6.242 0 016.25 6.25 6.242 6.242 0 01-6.25 6.25 6.242 6.242 0 01-6.25-6.25 6.242 6.242 0 016.25-6.25z"
          ></path>
        </g>
      </g>
    </SVGIcon>
  );
};
