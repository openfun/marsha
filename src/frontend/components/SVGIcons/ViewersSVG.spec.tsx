import React from 'react';
import renderer from 'react-test-renderer';
import { ViewersSVG } from './ViewersSVG';

it('renders ViewersSVG correctly', () => {
  const ViewersSVGSnapshot = renderer
    .create(
      <ViewersSVG
        baseColor={'blue-off'}
        height={'27.08'}
        hoverColor={'blue-active'}
        title={'Show viewers'}
        width={'45.83'}
      />,
    )
    .toJSON();
  expect(ViewersSVGSnapshot).toMatchSnapshot();
});
