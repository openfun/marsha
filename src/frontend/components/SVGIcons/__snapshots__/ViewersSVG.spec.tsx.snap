// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ViewersSVG correctly 1`] = `
.c0 path {
  fill: #81ade6;
}

.c0:hover path {
  fill: #035ccd;
}

<svg
  className="c0"
  height="27.08"
  viewBox="0 0 22 13"
  width="45.83"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Show viewers
  </title>
  <g
    fill="none"
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      transform="translate(-645 -927)"
    >
      <g
        transform="translate(50 691)"
      >
        <g
          transform="translate(594 231)"
        >
          <path
            d="M12 12c1.63 0 3.07.446 4.24 1.029 1.08.548 1.76 1.782 1.76 3.12v.708c0 .629-.45 1.143-1 1.143H7c-.55 0-1-.514-1-1.143v-.697c0-1.349.68-2.583 1.76-3.12A9.355 9.355 0 0112 12zm-7 1c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58A2.01 2.01 0 001 16.43V17c0 .55.45 1 1 1h3.5v-1.61c0-.83.23-1.61.63-2.29zM19 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85A6.95 6.95 0 0019 14c-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H22c.55 0 1-.45 1-1v-.57zM12 5c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;
